TipCatDotNet.Api/Models/Payments/Enums/AllowedPaymentMethods.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.ComponentModel;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace TipCatDotNet.Api.Models.Payments.Enums
+{
+    public static class AllowedPaymentMethods
+    {
+        public static List<string> ToList()
+        {
+            var t = typeof(PaymentMethods);
+            return Enum.GetValues(t).Cast<Enum>().Select(x => x.GetDescription()).ToList();
+        }
+
+
+        private static string GetDescription(this Enum value)
+        {
+            var type = value.GetType();
+            var name = Enum.GetName(type, value);
+            if (name != null)
+            {
+                var field = type.GetField(name);
+                if (field != null)
+                {
+                    var attr = Attribute.GetCustomAttribute(field, typeof(DescriptionAttribute)) as DescriptionAttribute;
+                    if (attr != null)
+                    {
+                        return attr.Description;
+                    }
+                }
+            }
+            return value.ToString();
+        }
+
+
+        private enum PaymentMethods
```

Comment:
If this enum is public, we may use it at the front-end

Reply:
By the way, why do we need exact these descriptions?

Reply:
This desctiptions needed to prettify values. I used it like a string value. This enum not needed to be use in front-end. They are allowed payments methods used by stripe intent to lend front-end's stripe.js component know which methods need to draw. It's proceed automaticaly in frond-end by calling : stripe.GetElements("client_secret");